giraffe/src/utils/getMargins.ts

```diff
@@ -11,18 +11,26 @@ export const getMargins = (
   showAxes: boolean,
   xAxisLabel: string,
   yAxisLabel: string,
-  yTicks: number[],
-  yTickFormatter: (tick: number) => string,
+  yTicks: string[] | number[],
+  yTickFormatter: (tick: number | string) => string,
   tickFont: string
 ) => {
   if (!showAxes) {
     return {top: 1, right: 1, bottom: 1, left: 1}
   }
 
-  const longestYTick = maxBy(
-    d => d.length,
-    yTicks.map(t => yTickFormatter(t))
-  )
+  let longestYTick
+  if (typeof yTicks[0] === 'string') {
+    longestYTick = maxBy(
+      d => d.length,
+      (yTicks as string[]).map(t => yTickFormatter(t))
+    )
+  } else {
+    longestYTick = maxBy(
+      d => d.length,
+      (yTicks as number[]).map(t => yTickFormatter(t))
+    )
+  }
```

Comment:
const formattedYTicks = yTicks.map(t => yTickFormatter(t))
  const longestYTick = maxBy(d => d.length, formattedYTicks)

Reply:
This should work!

Reply:
I'm not sure if I'm doing something wrong or missing some code elsewhere, but when I make the changes you suggested above I get 2 errors. It says map is not callable because the signatures are not compatible with each other (line 22), and it says property length does not exist on type 'unknown' (line 23). Do you have any ideas on why TS might be complaining on my version but not on yours?

Reply:
you need to also make the change I suggested above for it to work.  ` yTicks: string[] | number[]` to  `  yTicks: Array<string | number>`

Reply:
let me know if it still doesn't work

Reply:
Never mind. Sorry missed the above suggestion about changing the type. It works!